cmake/OpenCVFindLAPACK.cmake

```diff
@@ -51,6 +51,13 @@ macro(ocv_lapack_check)
     if(NOT "${OPENCV_CBLAS_H_PATH_${_lapack_impl}}" STREQUAL "${OPENCV_LAPACKE_H_PATH_${_lapack_impl}}")
       list(APPEND _lapack_content "#include \"${OPENCV_LAPACKE_H_PATH_${_lapack_impl}}\"")
     endif()
+    list(APPEND _lapack_content "
+#ifdef OPENCV_USE_LAPACK_PREFIX
+#define LAPACK_FUNC(f) LAPACK_##f
```

Comment:
`-DOPENCV_USE_LAPACK_PREFIX=LAPACK_`

It makes sense to define prefix in CMake.

---

Moreover, we don't need global compiler definition at all.
It is a generated file. So generate this content through CMake conditions.

Reply:
Ok will remove the global alltogether and generate based in CMake conditions